cfme/tests/cloud_infra_common/test_custom_attributes_rest.py

```diff
@@ -63,12 +64,37 @@ def _get_vm():
     return _get_vm
 
 
+@pytest.yield_fixture(scope='module')
+def get_service(appliance):
+    uid = fauxfactory.gen_alphanumeric(5)
+    name = 'test_rest_service_{}'.format(uid)
+
+    def _get_service():
+        service = appliance.rest_api.collections.services.find_by(name=name)
+        if not service:
+            body = {
+                'name': name,
+                'description': 'Test REST Service {}'.format(uid),
+            }
+            service = appliance.rest_api.collections.services.action.create(body)
+        return service[0]
+
+    yield _get_service
```

Comment:
Why do we yield here?

Reply:
@jamesooden So we're yielding the function here, and its called later down on L 145: 
https://github.com/ManageIQ/integration_tests/pull/6648/files#diff-82f0d8e84509575e71ce820205c41e29R145

The yield allows for cleanup after the test scope is completed. In this case, the fixture is module scoped, so after all of the test functions (parametrized) have executed, the module will close out by returning to here and executing the try/except block below to delete the service.